libusb/hid.c

```diff
@@ -548,11 +551,100 @@ int HID_API_EXPORT hid_exit(void)
 	return 0;
 }
 
+
+static void fill_device_info_for_device(libusb_device_handle *handle, struct hid_device_info * cur_dev, struct libusb_device_descriptor *desc)
+{
+	/* VID/PID */
+	cur_dev->vendor_id = desc->idVendor;
+	cur_dev->product_id = desc->idProduct;
+
+	/* Release Number */
+	cur_dev->release_number = desc->bcdDevice;
+
+	if (handle) {
```

Comment:
I always recommend not to have large (most) of the function body under an `if` condition, i.e. in this case:
```
if (!handle)
    return;
```
so the rest of the body doesn't have an extra indentation

---
yes, that is another holy war about function early returns, and if that bothers you a lot - just place the content of the `if` branch body into another function
the point - avoid extra indentation of the large amount of code

---
and yes - its been even worse before, but this time we're doing curated refactoring of this part, and not just "different people adding random things"